src/Orleans.Runtime.Abstractions/Streams/ISiloPersistentStreamConfigurator.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Orleans.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orleans.Streams
+{
+    public interface ISiloPersistentStreamConfigurator
+    {
+        ISiloPersistentStreamConfigurator Configure<TOptions>(Action<OptionsBuilder<TOptions>> configureOptions)
+          where TOptions : class, new();
+
+        ISiloPersistentStreamConfigurator ConfigureComponent<TOptions, TComponent>(Action<OptionsBuilder<TOptions>> configureOptions, Func<IServiceProvider, string, TComponent> factory)
```

Comment:
nit:  As configuring options in the ConfigureComponent is optional, seems like listing the factory first then the "optional" options would make the call pattern more consistent.

Reply:
agreed. And I'm thinking move this method to extension . Since this method are really just calling the other two methods